msgpack-core/src/main/java/org/msgpack/core/MessagePack.java

```diff
@@ -37,406 +37,100 @@
-    {
-        private boolean readStringAsBinary = true;
-        private boolean readBinaryAsString = true;
-
-        private CodingErrorAction onMalFormedInput = CodingErrorAction.REPLACE;
-        private CodingErrorAction onUnmappableCharacter = CodingErrorAction.REPLACE;
-
-        private int maxUnpackStringSize = Integer.MAX_VALUE;
-        private int stringEncoderBufferSize = 8192;
-        private int stringDecoderBufferSize = 8192;
-        private int packerBufferSize = 8192;
-        private int packerSmallStringOptimizationThreshold = 512; // This parameter is subject to change
-        private int packerRawDataCopyingThreshold = 512;
-
-        public Config build()
-        {
-            return new Config(
-                    readStringAsBinary,
-                    readBinaryAsString,
-                    onMalFormedInput,
-                    onUnmappableCharacter,
-                    maxUnpackStringSize,
-                    stringEncoderBufferSize,
-                    stringDecoderBufferSize,
-                    packerBufferSize,
-                    packerSmallStringOptimizationThreshold,
-                    packerRawDataCopyingThreshold
-            );
-        }
-
-        public ConfigBuilder readStringAsBinary(boolean enable)
-        {
-            this.readStringAsBinary = enable;
-            return this;
-        }
-
-        public ConfigBuilder readBinaryAsString(boolean enable)
-        {
-            this.readBinaryAsString = enable;
-            return this;
-        }
-
-        public ConfigBuilder onMalFormedInput(CodingErrorAction action)
-        {
-            this.onMalFormedInput = action;
-            return this;
-        }
-
-        public ConfigBuilder onUnmappableCharacter(CodingErrorAction action)
-        {
-            this.onUnmappableCharacter = action;
-            return this;
-        }
-
-        public ConfigBuilder maxUnpackStringSize(int size)
-        {
-            this.maxUnpackStringSize = size;
-            return this;
-        }
-
-        public ConfigBuilder stringEncoderBufferSize(int size)
-        {
-            this.stringEncoderBufferSize = size;
-            return this;
-        }
-
-        public ConfigBuilder stringDecoderBufferSize(int size)
-        {
-            this.stringDecoderBufferSize = size;
-            return this;
-        }
-
-        public ConfigBuilder packerBufferSize(int size)
-        {
-            this.packerBufferSize = size;
-            return this;
-        }
-
-        public ConfigBuilder packerSmallStringOptimizationThreshold(int threshold)
-        {
-            this.packerSmallStringOptimizationThreshold = threshold;
-            return this;
-        }
-
-        public ConfigBuilder packerRawDataCopyingThreshold(int threshold)
-        {
-            this.packerRawDataCopyingThreshold = threshold;
-            return this;
-        }
-    }
+    private static MessagePackFactory defaultFactory = new MessagePackFactory();
```

Comment:
Storing default factory class as a mutable static variable doesn't seem a good idea. For instance, if a dependent library uses MessagePack and internally sets a customized factory, it'll unexpectedly affect the application that uses MessagePack.